openquakeplatform_taxtweb/utils/taxtweb_eng.py

```diff
@@ -0,0 +1,4037 @@
+#!/usr/bin/env python
+import sys, math, re, io
+
+from openquakeplatform_taxtweb.utils.taxtweb_maps import (
+    material, date_type, occu_type, bupo_type, plsh_type, stir_type,
+    plan_irre, plan_seco, vert_irre, vert_seco, wall_type, roof_shap,
+    roof_cove, roof_mate, roof_conn, floo_syma, floo_syty, foun_type,)
+from openquakeplatform_taxtweb.utils.taxtweb_head import (mat_tech, mat_tead, mat_prop, llrs_type,
+                                llrs_duct, occu_spec, roof_sys, floo_conn)
+from openquakeplatform_taxtweb.utils.taxonomy import taxonomy_short2full, Ret
+
+taxonomy = None
+
+def is_not_negative_int(s):
+    try:
+        if int(s) < 0 or int(s) != float(s):
+            return False
+        else:
+            return True
+    except ValueError:
+        return False
+
+def is_not_negative_float(s):
+    try:
+        if float(s) < 0.0:
+            return False
+        else:
+            return True
+    except ValueError:
+        return False
+
+def is_in_rect_angle_float(s):
+    if not is_not_negative_float(s):
+        return False
+
+    if float(s) > 90.0:
+        return False
+    return True
+
+def is_or_are_given(n):
+    return n + (" is" if n <= 1 else " are") + " given."
+
+class TaxtSel(object):
+    def __init__(self, name, items=[], val=-1, disabled=False, change_cb=None):
+        '''
+        items => list of 1 element dicts
+        val => int identifing current val item
+        '''
+        self._name = name
+        self._items = items[:]
+        if items:
+            if val == -1:
+                self._val = 0
+            else:
+                self._val = val
+        else:
+            self._val = val
+
+        self._disabled = disabled
+        self._first_disabled = False
+        self._change_cb = change_cb
+
+    def empty(self):
+        self._items = []
+        self._val = -1
+
+    def disabled(self, disabled=None):
+        if disabled is None:
+            return self._disabled
+        else:
+            self._disabled = disabled
+
+    def first_disabled(self, first_disabled=None):
+        if first_disabled is not None:
+            self._first_disabled = first_disabled
+
+        return self._first_disabled
+
+    def items(self, items=None, val=0):
+        '''
+        items => list of 1 element dicts
+        val => int identifing current val item
+        '''
+        if items == None:
+            return self._items
+        else:
+            self._items = items[:]
+
+        self.val(val)
+
+    def val(self, val=-1):
+        if val != -1:
+            if isinstance(val, str):
+                val = self.items.index(val)
```

Comment:
This is dangerous in the context of porting to Python 3. What is val? A `bytes` or a `unicode` object?